### PR TITLE
fix: remove hmr port

### PR DESCRIPTION
**Pull request by jaulz**
If the `hmr` endpoint uses the same port, we cannot use the Vite server in SSR middleware mode because the port is already blocked.

```js
  const vite = await createServer({
    server: { middlewareMode: 'ssr' }
  })
```

### CODE REVIEW

**Comment by innocenzi** [comment]
Can you ELI5 the issue? I never used SSR mode before. The docs says:

> When using `server.middlewareMode` and `server.https`, setting `server.hmr.server` to your HTTPS server will process HMR secure connection requests through your server. This can be helpful when using self-signed certificates.

Is it related to your issue?

Also, have you checked that this change does't break other workflows (normal HMR with and without `https`)?

**Comment by jaulz** [comment]
Sorry, I missed that we can simply pass to the configuration to override the default (same) port:
```
  server: {
    hmr: {
      port: undefined,
    }
  },
```

With SSR you basically don't use the `vite` server itself but instead you only use its middleware:
https://vitejs.dev/guide/ssr.html
